src/sentry/sentry_metrics/consumers/indexer/processing.py

```diff
@@ -1,18 +1,28 @@
 import logging
 
 from arroyo.types import Message
+from django.conf import settings
 
-from sentry.sentry_metrics import indexer
-from sentry.sentry_metrics.configuration import UseCaseKey
+from sentry.sentry_metrics.configuration import IndexerStorage, MetricsIngestConfiguration
 from sentry.sentry_metrics.consumers.indexer.batch import IndexerBatch
 from sentry.sentry_metrics.consumers.indexer.common import MessageBatch
+from sentry.sentry_metrics.indexer.cloudspanner.cloudspanner import CloudSpannerIndexer
+from sentry.sentry_metrics.indexer.mock import MockIndexer
+from sentry.sentry_metrics.indexer.postgres.postgres_v2 import PostgresIndexer
 from sentry.utils import metrics
 
 logger = logging.getLogger(__name__)
 
+STORAGE_TO_INDEXER = {
+    # TODO(add cloudspanner options)
+    IndexerStorage.CLOUDSPANNER: lambda: CloudSpannerIndexer(**settings.CLOUDSPANNER_OPTIONS),
+    IndexerStorage.POSTGRES: PostgresIndexer(),
+    IndexerStorage.MOCK: MockIndexer(),
+}
+
```

Comment:
```suggestion
_MOCK = MockIndexer()
_POSTGRES = PostgresIndexer()

STORAGE_TO_INDEXER = {
    # TODO(add cloudspanner options)
    IndexerStorage.CLOUDSPANNER: lambda: CloudSpannerIndexer(**settings.CLOUDSPANNER_OPTIONS),
    IndexerStorage.POSTGRES: lambda: _POSTGRES,
    IndexerStorage.MOCK: lambda: _MOCK,
}

```

otherwise this will not work, because some of your dict values are callables that return indexers (cloud spanner), while others are actual instances (pg/mock)

Reply:
oh wait, I don't think cloudspanner needs to be a lamda tho right? the options are static

Reply:
yeah but they may not be set, right? if you made it static, having a valid cloudspanner config is now unconditionally required

Reply:
yeah yeah you are le correct, updated with your suggestion :)